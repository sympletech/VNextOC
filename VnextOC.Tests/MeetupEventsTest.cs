using ApprovalTests.Reporters;
using ApprovalUtilities.Utilities;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.IO;
using VnextOC.MeetupApi;
using Approvals = ApprovalTests.Approvals;

// http://api.meetup.com/2/events?group_id=2983232&key=0393c247c5f1443107c312142206710

namespace VnextOC.Tests
{
    [TestClass]
    [UseReporter(typeof(DiffReporter))]
    public class MeetupEventsTest
    {
        [TestMethod]
        public void TestJsonEventTransformation()
        {
            var json = File.ReadAllText(PathUtilities.GetAdjacentFile("SampleEventData.json"));

            var events = Meetup.ParseEvent(json);
            
            Approvals.VerifyAll(events, "event");
        }

        [TestMethod]
        public void TestParseNamePerson()
        {
            var name = "Paul D. Sheriff - Architecting Applications for Multiple-User-Interfaces";

            var e = new Event().ParseName(name);

            Approvals.Verify(e);
        }
    }
}
