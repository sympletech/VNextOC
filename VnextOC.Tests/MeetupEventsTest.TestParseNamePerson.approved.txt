Event
{
	Title:  Architecting Applications for Multiple-User-Interfaces
	Speaker: Paul D. Sheriff 
	Time: 1/1/0001 12:00:00 AM
	Description: NULL
	EventUrl: NULL
	Address: NULL
}
